proto/pocket/supplier/tx.proto

```diff
@@ -1,7 +1,22 @@
 syntax = "proto3";
+
 package pocket.supplier;
 
 option go_package = "pocket/x/supplier/types";
 
 // Msg defines the Msg service.
-service Msg {}
\ No newline at end of file
+service Msg {
+  rpc SubmitProof (MsgSubmitProof) returns (MsgSubmitProofResponse);
+}
+message MsgSubmitProof {
```

Comment:
We should revisit `MsgSubmitProof` params to comply with @h5law 's changes with `VerifyClosestProof` which has the following signature [1]: 

```go
func VerifyClosestProof(proof *SparseMerkleClosestProof, root []byte, spec *TreeSpec) (bool, error)
```

And uses `SparseMerkleClosestProof` obtained from the rewritten`ProveClosest` which has the following structure [1]:

```go
type SparseMerkleClosestProof struct {
	Path             []byte             // the path provided to the ProveClosest method
	FlippedBits      []int              // the index of the bits flipped in the path during tree traversal
	Depth            int                // the depth of the tree when tree traversal stopped
	ClosestPath      []byte             // the path of the leaf closest to the path provided
	ClosestValueHash []byte             // the value hash of the leaf closest to the path provided
	ClosestProof     *SparseMerkleProof // the proof of the leaf closest to the path provided
}
```

[1] https://github.com/pokt-network/smt/blob/main/smt_proofs_test.go#L288
[2] https://github.com/pokt-network/smt/blob/main/proofs.go#L166C1-L173C2

Test: https://github.com/pokt-network/smt/blob/main/smt_proofs_test.go#L288

Reply:
Yeah @red-0ne is correct, this new struct includes all the values previously returned from the `ProveClosest` method but also links them directly to the path, the `VerifyClosestProof` ensures the proof is structured in a valid way and passes the relevent checks to ensure the path does produce the closestPath found. In addition to this we should, when verifying the proof, ensure that `proof.ClosestPath == {what we expected it to be}` and then the verification of the proof with ensure that path definietly produced the one found and then also validate that proof in turn.

We can continue to embed this entire struct in the protobufs using `proof.Marshal()` and store it as a `[]byte`

Reply:
@h5law we got confused by this part
> proof.ClosestPath == {what we expected it to be}
Did you mean `proof.Path` ? which is what is provided to `ProveClosest` ?

Reply:
@red-0ne yes! sorry my error the `proof.Path` is what will be determined by the function of on-chain data (blockhash etc) **not** the `proof.ClosestPath` <- this is the path we found. Good 👀